src/logger/src/lib.rs

```diff
@@ -32,3 +32,17 @@ fn extract_guard<G>(lock_result: LockResult<G>) -> G {
         Err(poisoned) => poisoned.into_inner(),
     }
 }
+
+pub fn update_metric_with_elapsed_time(
+    metric: &SharedMetric,
+    start_time_us: u64,
+    action_to_log: &str,
+    thread_name: &str,
+) {
+    let delta_us = utils::time::get_time_us(utils::time::ClockType::Monotonic) - start_time_us;
+    info!(
```

Comment:
keep this the log level consistent, you currently have a mix of `info!` and `debug!`

Reply:
`debug!` is only used for printing the total API time for each request (not related to the snapshot ones). I used `info!` in the other places in case the user is interested in those durations (at the `Info` level) and doesn't want to configure the metrics (also for `create` requests you will have all those durations without calling `FlushMetrics` more times). Do you think `debug!` would be more appropriate for these too?
In case we decide for `debug!` everywhere, I can update all the log messages to not be prettified, for `info!` on the other hand I think those sentences should sound somehow right/clear.

Reply:
Production scenarios most likely use metrics and not logs to measure these things - logs don't scale well. Even so, I'm not sure what's the right level for these log messages... Both `debug!` and `info!` make sense, so I leave it to your personal choice.

> In case we decide for debug! everywhere, I can update all the log messages to not be prettified, for info! on the other hand I think those sentences should sound somehow right/clear.

I partially disagree. _Clear_ messages should always be the case, _pretty_ messages are just nice to have - but not at the expense of other things. I'm happy with clear even pretty messages as long as they can be kept short. I believe a small binary and small logfile footprint is much more appreciated than highly descriptive log messages.

Reply:
'create full snapshot' VMM action took 2754442 us.
'create full snapshot' API request took 2755308 us.
Shorter than this doesn't seem clear to me. Unfortunately, sometimes is actually hard to be (very) clear without lot of words :(.